UOP1_Project/Assets/Scripts/Pool/Pool.cs

```diff
@@ -0,0 +1,63 @@
+﻿using OP1.Factory;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace OP1.Pool
+{
+	/// <summary>
+	/// A generic pool that generates members of type T on-demand via a factory.
+	/// </summary>
+	/// <typeparam name="T">Specifies the type of elements to pool.</typeparam>
+	public abstract class Pool<T> : ScriptableObject, IPool<T> where T : IPoolable
+	{
+		protected readonly Stack<T> _available = new Stack<T>();
+		public abstract IFactory<T> Factory { get; set; }
+
+		public virtual T Add()
+		{
+			return Factory.Create();
+		}
```

Comment:
I don't get this method. It's called `Add` but it doesn't add the new instance to the pool? And I'm thinking maybe you meant to make it protected, not public. It would be very misleading to have a public `Add` method that returns a new instance just like `Request`. What's the difference, and, since it returns an instance, does that mean I can use it, or -as the name suggests- is it added to the pool and will I have problems if use it and then call `Request` and it hands me that same instance?

I see that you override it in the `ComponentPool` to set the parent when creating a new instance, so the idea is to be able to initialize the object when first instantiating it. Makes sense, but remember to make it protected, and I'd change the name to `Create` since it calls `Factory.Create()`.

Reply:
I had it as Create. Not sure what I was thinking when I renamed. Makes more sense as Create.